docs/design/repair-with-hashes.md

```diff
@@ -0,0 +1,63 @@
+
+Failed hash, signature, or public keys checks may indicate cheating. Therefore audit must implement this same piece hash check logic. Changes to audit and the reporting of cheaters are otherwise outside the scope of this document.  
+
+## Rationale
+
+Both download and Reed Solomon decoding create more load for a satellite than hashing, so hashing should be preferred over download even a single more piece.
+
+Downloading for repair is significantly different enough from streaming as to warrant a new, custom implementation.
+
+Using only the minimum number of pieces means that Reed-Solomon does not act as a check during repair. Hence hashing is used instead. While and uplink could potentially send signed bogus data to a storage node, the storage node would not be penalized by these actions. This requires that Audit implements a similar piece hash check instead of relying solely on Reed-Solomon encoding.
+
+The size of all piece hashes downloaded should be roughly equal to a segment size : 64mB.  It seems preferable to keep this in memory over dealing with persistance to disk.
+
+## Implementation
+
+1. Add an optional `PieceHash` and `OrderLimit` fields to the `PieceDownloadResponse` protocol buffer for returning hashes.
+
+2. Alter the storage node code to populate `PieceHash` and `OrderLimit` when the `OrderLimit` `Action` is `GET_REPAIR`.
+
+3. Write code which can validate a piece using a `PieceHash` and `OrderLimit` as described in design.
+
+4. Write code which ensures that enough pieces share the same `PiecePublicKey` as described in design.
+
+5. Impelement custom repair download code which uses a Reed-Solomon minimum number of pieces, persisting the downloaded pieces only in memory. If statistics are available, prefer to download from faster storage nodes using the power of two choices.
+
+6. Wire in validation logic to repair loop.  If a piece fails these checks, download another.
+
+7. Reassemble the segment using Reed-Solomon decoding and continue with repair as previously implemented.
+
+## Open issues
+
+With all this hashing and checking of signatures, are we still sure this method is preferred?
```

Comment:
The amount of data transferred is still reduced with this method for repair, even with the extra hashing/checking logic. And since bandwidth is the most expensive resource, it makes sense to prioritize reducing that over having a "simpler" solution.